lambdas API review (#226)
decentraland/adr: content/ADR-208-lambdas-api-review.md

---
layout: adr

> adr: 208

Reviewer: Should this number be the id of the PR?

title: Lambdas API Review
date: 2020-02-20
status: Draft
type: Standards Track
spdx-license: CC0-1.0
authors:
  - hugoArregui
---

## Abstract

> Catalyst's Lambdas is a service created to support many unrelated operations needed originally by the reference implementation client, but as time goes by, new, more efficient endpoints or ways of performing the same operation are implemented, but the service still contains every one of this iterations. The idea of this ADR is to describe the status of those endpoints in order to be able to clean up the service and have a more robust catalyst implementation.

Reviewer: ```suggestion
Catalyst's Lambdas is a service created to support many unrelated operations needed initially by the reference implementation client. As time goes by, new and more efficient endpoints with the same responsibilities are implemented, but the service still contains every iteration. The idea of this ADR is to describe those endpoints statuses to clean up the service, and have a more robust catalyst implementation.
```

## Context, Reach & Prioritization

> Given the nature of the service lambdas is one of the most complex pieces of software inside the catalyst node, documenting the changes and been able to cleanup the service drasticaly will improve the mantenability and efficence of the catalyst itself.

Reviewer: ```suggestion
Given its nature, the Lambdas service is one of the most complex pieces of software inside the catalyst node. Documenting the changes and being able to clean up the service drastically will improve the maintainability and efficiency of the catalyst itself.
```

## Specification

### GET /lambdas/health

Action: deletion

> Lambdas health endpoint has been replaced for the realm description endpoint `/about` specified by [ADR-110: Realm description](./ADR-110-realm-description.md).

Reviewer: ```suggestion
Lambdas' health endpoint has been replaced with the realm description endpoint `/about` specified by [ADR-110: Realm description](./ADR-110-realm-description.md).
```

### GET /lambdas/contentv2/parcel_info

Action: deletion

> This endpoint is just a proxy to content-server's `POST /entities/active'.

Reviewer: ```suggestion
This endpoint is a proxy to the content-server's `POST /entities/active'.
```

### GET /lambdas/contentv2/contents/:cid

Action: deletion

> This endpoint is just a proxy to content-server's `GET /contents/:hashId'.

Reviewer: ```suggestion
This endpoint is a proxy to the content-server's `GET /contents/:hashId'.
```

### GET /lambdas/scenes

Action: deletion

> This endpoint is just a proxy to content-server's `POST /entities/active'.

Reviewer 2: ```suggestion
This endpoint is just a proxy to content-server's `POST /entities/active`.
```

Reviewer: ```suggestion
This endpoint is a proxy to the content-server `POST /entities/active`.
```

### GET /lambdas/profiles

Action: deletion

> This endpoint is replaced by `POST /lambdas/profiles`, this endpoint had a limit on the number of profiles you can request at once given by the lenght of the url, the POST version of this endpoint removes this restriction.

Reviewer: ```suggestion
This endpoint is replaced by `POST /lambdas/profiles`. It had a limit on the number of profiles you can request at once given by the URL length. The POST version of this endpoint removes this restriction.
```

> ### GET /lambdas/profiles/{id}

Reviewer 2: ```suggestion
### GET /lambdas/profiles/{id}
```

This endpoint was used to leverage the caching on the edge servers and CDN

Action: deletion

This endpoint is replaced by `POST /lambdas/profiles`, in which many profiles can be requested at once.

### GET /lambdas/contracts/servers

Action: deletion

> This endpoint is a proxy to the catalyst smart contract, this can be replaced by a call to the contract. To make things easier for clients, a new version of the calyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the data of the contract at the time of the release.

Reviewer: ```suggestion
This endpoint is a proxy to the catalyst's smart contract that can be replaced by calling it directly. To make things easier for clients, a new version of the catalyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the contract data at the time of the release.
```

### GET /lambdas/contracts/pois

> Action: deletion

Reviewer: ```suggestion
This endpoint is a proxy to the POIs' smart contract that can be replaced by calling it directly. To make things easier for clients, a new version of the catalyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the contract data at time of the release.
```

> This endpoint is a proxy to the POIs smart contract, this can be replaced by a call to the contract. To make things easier for clients, a new version of the calyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the data of the contract at the time of the release.

Reviewer: ```suggestion
This endpoint is a proxy to the POIs' smart contract that can be replaced by calling it directly. To make things easier for clients, a new version of the catalyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the contract data at the time of the release.
```

> ### GET /lambdas/contracts/denylisted-names

Reviewer 2: Requesting any of these lists to the ethereum nodes takes more than 30seconds. The client uses the `/lambdas/contracts/servers` to startup, and delaying the initial loading by 30sec is a no-go from a UX point of view. This should ONLY be enacted once there is a /pick-realm endpoint somewhere that returns a realm in within a couple milliseconds CC @aleortega

Author: The idea was to use catalyst-client with a snapshot of the content of the contract

Reviewer 2: ok... that sounds hardcoded but I think it works.. can you fast-forward adding an endpoint somewhere to have the realm selection server-side? I think that would solve many issues

Author: awesome! I'm resolving this conversation and DM you about realm picking, so we don't mix the realm picking here, thanks!

Action: deletion

> This endpoint is a proxy to the names denylist smart contract, this can be replaced by a call to the contract. To make things easier for clients, a new version of the calyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the data of the contract at the time of the release.

Reviewer: ```suggestion
This endpoint is a proxy to the names denylist smart contract that can be replaced by calling it directly. To make things easier for clients, a new version of the catalyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the contract data at the time of the release.
```

This endpoint is a proxy to the catalyst's smart contract that can be replaced by calling it directly. To make things easier for clients, a new version of the catalyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the contract data at the time of the release.

### GET /lambdas/contracts/pois

Action: deletion

This endpoint is a proxy to the POIs' smart contract that can be replaced by calling it directly. To make things easier for clients, a new version of the catalyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the contract data at the time of the release.

### GET /lambdas/contracts/denylisted-names

Action: deletion

This endpoint is a proxy to the names denylist smart contract, this can be replaced by a call to the contract. To make things easier for clients, a new version of the calyst-client will be released, including both a way to query the contract if you have a web3 provider and a snapshot of the data of the contract at the time of the release.

### GET /lambdas/collections/wearables-by-owner/{address}

Action: deletion

> Replaced by GET /lambdas/nfts/wearables/{address}, which returns the same data, but its paginated.

Reviewer: ```suggestion
Replaced by GET /lambdas/nfts/wearables/{address}, which returns the same data, but paginated.
```
